service/history/workflow/workflow_task_state_machine.go

```diff
@@ -401,19 +409,25 @@ func (m *workflowTaskStateMachine) AddWorkflowTaskStartedEvent(
 
 	// Avoid creating new history events when workflow tasks are continuously failing
 	if workflowTask.Attempt == 1 {
+		workflowTask.SuggestContinueAsNew, workflowTask.HistorySizeBytes = m.getHistorySizeInfo()
+
```

Comment:
Why not to compute it for every attempt?

Reply:
If we compute it for every attempt, but only actually write the event to history on the first attempt, then the value used by the workflow may be different from the value written to history, so replay would cause a determinism error. Like 90% of the complexity of this PR is just for that

Reply:
For the record, the answer here is that we write a new started event one when the retried wft completes, so as long as that new started event has the right values, we're good